grails-app/services/org/pih/warehouse/stockTransfer/StockTransferService.groovy

```diff
@@ -211,6 +210,18 @@ class StockTransferService {
         return deleteOrderItem(orderItem.order, orderItem)
     }
 
+    Order deleteAllStockTransferItems(String id) {
+        Order order = Order.get(id)
+        if (!order) {
+            throw new IllegalArgumentException("No stockTransfer found with ID ${id}")
+        }
+
+        def itemsToRemove = order.orderItems?.findAll { it }
```

Comment:
why you need the findAll here?

Reply:
It is a workaround to deal with `ConcurrentModificationException`